frontend/src/metabase/sharing/components/EmailAttachmentPicker.jsx

```diff
@@ -0,0 +1,206 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+
+import ButtonGroup from "metabase/components/ButtonGroup";
+import CheckBox from "metabase/components/CheckBox";
+import StackedCheckBox from "metabase/components/StackedCheckBox";
+import Toggle from "metabase/components/Toggle";
+
+export default class EmailAttachmentPicker extends Component {
+  state = {
+    attachmentType: null,
+    selectedCardIds: new Set(),
+  };
+
+  static propTypes = {
+    pulse: PropTypes.object.isRequired,
+    setPulse: PropTypes.func.isRequired,
+  };
+
+  DEFAULT_ATTACHMENT_TYPE = "csv";
+
+  updateStateFromCards = () => {
```

Comment:
Not worth changing, but just FYI that this could be a normal method since it's always invoked as `this. updateStateFromCards()`